src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,596 @@
+      const id = random.number().toString();
+
+      const bufferObservable = fetchRecordEpic.createBufferObservable(id);
+
+      let action;
+      bufferObservable
+        .take(1)
+        .subscribe(a => action = a);
+
+      return setTimeoutPromise(() => {
+        expect(action).to.deep.equal({
+          type: "PUSH_RECORD",
+          payload: new FakeModel({
+            id,
+            fullText: expectedResult.fullText,
+          }),
+        });
+      }, DEFAULT_COALESCE_BUFFER_TIME + 1);
+    });
+
+  });
+
+  describe("performBufferedRequest", () => {
+
+    it("should return separate buffered observables for different fetchRecordEpic instances", () => {
+      const firstId = random.number().toString();
+      const secondId = random.number().toString();
+      const timeoutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      const firstPerformRequest = fetchRecordEpic.performBufferedRequest(firstId);
+      const newContext = {
+        ...context,
+        name: random.word(),
+      };
+      const newFetchRecordEpic = new FetchRecordEpic(newContext);
+      const secondPerformRequest = newFetchRecordEpic.performBufferedRequest(secondId);
+
+      let firstAction;
+      firstPerformRequest
+        .subscribe(a => firstAction = a);
+
+      let secondAction;
+      secondPerformRequest
+        .subscribe(a => secondAction = a);
+
+      return setTimeoutPromise(() => {
+        expect(
+          firstAction,
+        ).to.not.deep.equal(secondAction);
+      }, timeoutPeriod);
+    });
+
+    it("should subscribe to a new buffered observable when requested after the timeout period", () => {
+      const id = random.number().toString();
+      const secondId = random.number().toString();
+      const firstTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 1;
+      const secondTimeOutPeriod = firstTimeOutPeriod * 2;
+
+      const createBufferObservableStub = stub(fetchRecordEpic, "createBufferObservable").returns(observable);
+
+      fetchRecordEpic.performBufferedRequest(id)
+        .subscribe();
+
+      setTimeout(() => {
+        fetchRecordEpic.performBufferedRequest(secondId)
+          .subscribe();
+      }, firstTimeOutPeriod);
+
+      return setTimeoutPromise(() => {
+        expect(
+          createBufferObservableStub.secondCall.args[0],
+        ).to.equal(secondId);
+      }, secondTimeOutPeriod);
+    });
+
+    it("should complete the buffered observable to prevent memory leaks after it emits once and the timeout passes", () => {
+      const id = random.number().toString();
+
+      const spies = {
+        next: spy(),
+        complete: spy(),
+      };
+
+      stub(fetchRecordEpic, "createBufferObservable").returns(observable);
+      fetchRecordEpic.performBufferedRequest(id)
+        .subscribe(spies);
+
+      return setTimeoutPromise(() => {
+        expect(
+          spies.complete.callCount,
+        ).to.equal(1, "it should complete immediately");
+      }, DEFAULT_COALESCE_BUFFER_TIME + 1);
```

Comment:
This still say "it should complete immediately", even though we are really expected it to complete after the timeout. Did you mean to change it then resolve the previous comment, or do you not think it should be changed?

Reply:
Oh, I didn't ignore your comment on purpose for sure.   I think it just got lost in the shuffle when I was making changes.  Thanks for reminding me.